internal/services/tesla_fleet_api_service.go

```diff
@@ -0,0 +1,141 @@
+	GetVehicles(token, region string) ([]TeslaVehicle, error)
+}
+
+var teslaScopes = []string{"openid offline_access", "user_data", "vehicle_device_data", "vehicle_cmds", "vehicle_charging_cmds", "energy_device_data", "energy_device_data", "energy_cmds"}
+
+type GetVehiclesResponse struct {
+	Response []TeslaVehicle `json:"response"`
+}
+
+type TeslaFleetAPIError struct {
+	Error            string `json:"error"`
+	ErrorDescription string `json:"error_description"`
+	ReferenceID      string `json:"ReferenceID"`
+}
+
+type TeslaAuthCodeResponse struct {
+	AccessToken  string `json:"access_token"`
+	RefreshToken string `json:"refresh_token"`
+	IDToken      string `json:"id_token"`
+	ExpiresIn    int    `json:"expires_in"`
+	State        string `json:"state"`
+	TokenType    string `json:"token_type"`
+}
+
+type teslaFleetAPIService struct {
+	Settings   *config.Settings
+	HTTPClient *http.Client
+	log        *zerolog.Logger
+}
+
+func NewTeslaFleetAPIService(settings *config.Settings, logger *zerolog.Logger) TeslaFleetAPIService {
+	return &teslaFleetAPIService{
+		Settings:   settings,
+		HTTPClient: &http.Client{},
+		log:        logger,
+	}
+}
+
+// CompleteTeslaAuthCodeExchange godoc
+// @Description Call Tesla Fleet API and exchange auth code for a new auth and refresh token
+// @Param       authCode - authorization code to exchange
+// @Param       redirectURI - redirect uri to pass on as part of the request to for oauth exchange
+// @Param       region - API region which is used to determine which fleet api to call
+// @Success     200 {object} services.TeslaAuthCodeResponse
+func (t *teslaFleetAPIService) CompleteTeslaAuthCodeExchange(authCode, redirectURI, region string) (*TeslaAuthCodeResponse, error) {
+	conf := oauth2.Config{
+		ClientID:     t.Settings.Tesla.ClientID,
+		ClientSecret: t.Settings.Tesla.ClientSecret,
+		Endpoint: oauth2.Endpoint{
+			TokenURL: t.Settings.Tesla.TokenAuthURL,
+		},
+		RedirectURL: redirectURI,
+		Scopes:      teslaScopes,
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
+	defer cancel()
+
+	tok, err := conf.Exchange(ctx, authCode, oauth2.SetAuthURLParam("audience", fmt.Sprintf("https://fleet-api.prd.%s.vn.cloud.tesla.com", region)), oauth2.SetAuthURLParam("grant_type", "authorization_code"))
+	if err != nil {
+		var e *oauth2.RetrieveError
+		errors.As(err, &e)
+		return nil, fmt.Errorf("error occurred completing authorization: %s", e.ErrorDescription)
+	}
+
+	return &TeslaAuthCodeResponse{
+		AccessToken:  tok.AccessToken,
+		RefreshToken: tok.RefreshToken,
+		ExpiresIn:    int(tok.Expiry.Unix()),
+		TokenType:    tok.TokenType,
+	}, nil
+}
+
+// GetVehicles godoc
+// @Description Call Tesla Fleet API to get a list of vehicles using authorization token
+// @Param       token - authorization token to be used as bearer token
+// @Param       region - API region which is used to determine which fleet api to call
+// @Success     200 {object} []services.TeslaVehicle
```

Comment:
These are docs for routes, which these are not—I wouldn't say that a _function_ had response code 200. Just say what the functions do.